src/overlays/actors/ovl_Bg_Ydan_Hasi/z_bg_ydan_hasi.c

```diff
@@ -21,23 +27,177 @@ const ActorInit Bg_Ydan_Hasi_InitVars = {
     (ActorFunc)BgYdanHasi_Update,
     (ActorFunc)BgYdanHasi_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/BgYdanHasi_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/BgYdanHasi_Destroy.s")
+static InitChainEntry sInitChain[] = {
+
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+extern Gfx* D_808BEC24[] = { 0x06007508, 0x06005DE0, 0x06005018 };
+extern UNK_TYPE D_06005780;
+extern UNK_TYPE D_06007798;
+extern Gfx* D_06005DE0;
+
+void BgYdanHasi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s16 pad1;
+    BgYdanHasi* this = THIS;
+    s32 localConst;
+    WaterBox* waterBox;
+    s16 pad2;
+
+    localConst = 0;
+    Actor_ProcessInitChain(thisx, sInitChain);
+    this->unk_168 = (u8)((thisx->params >> 8) & 0x3F);
+    thisx->params = thisx->params & 0xFF;
+    waterBox = globalCtx->colCtx.stat.colHeader->waterBoxes + 0x1;
+    DynaPolyInfo_SetActorMove(&this->dyna, 1);
+    if (thisx->params == 1) { // Water the moving platform floats on in B1. Never runs in Master Quest
+        thisx->initPosRot.pos.y = (thisx->initPosRot.pos.y + -5.0f);
+        thisx->posRot.pos.y = thisx->initPosRot.pos.y;
+        waterBox->unk_02 = thisx->initPosRot.pos.y;
+        this->actionFunc = BgYdanHasi_initWater;
+    } else {
+        if (thisx->params == 0) { // Moving platform on the water in B1
```

Comment:
`else if` here imo

Reply:
Doesn't match when using `else if`

Reply:
Oh nvm, I missed the line at the end. Unfortunate.